### PR TITLE
Jingwei/add first last name field updatehook

### PR DESCRIPTION
**Pull request by sky1122**
the minimum length of the Firstname field is 1, and Lastname can be null.
also adding before update hook to firstName and lastName similar to the username.

### CODE REVIEW

File: extensions/users-permissions/models/User.js [hunk 1]
```diff
@@ -5,9 +5,24 @@ module.exports = {
...
   lifecycles: {
+    // async beforeUpdate(params, data) {
+    //   if (data.username)
+    //     data.username = data.username.trim().replace(/\s+/g, '_');
+    // },
     async beforeUpdate(params, data) {
-      if (data.username)
+      const {username, firstName, lastName} = data;
+      if (username !== null && username !== undefined) {
+        strapi.log.info('username');
```
Comment (chungthuang) [inline extensions/users-permissions/models/User.js:15]: Can we remove the debug logs?

File: extensions/users-permissions/models/User.js [hunk 2]
```diff
@@ -5,9 +5,24 @@ module.exports = {
...
    */
   lifecycles: {
+    // async beforeUpdate(params, data) {
+    //   if (data.username)
+    //     data.username = data.username.trim().replace(/\s+/g, '_');
+    // },
     async beforeUpdate(params, data) {
-      if (data.username)
+      const {username, firstName, lastName} = data;
+      if (username !== null && username !== undefined) {
```
Comment (chungthuang) [inline extensions/users-permissions/models/User.js:14]: You can simplify this check to https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in.

**Comment by chungthuang** [comment]
We are migrating from this repo. Can you also update https://github.com/dev-launchers/strapiv4?

**Comment by sky1122** [comment]
Yeah, I will do that too.

**Review by chungthuang — APPROVED** [review]

**Review by pyxld-kris — APPROVED** [review]

**Pull request closed without merging**